src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/directionsquestionnaire/HearingContentProvider.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content.directionsquestionnaire;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.ExpertReportContent;
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.HearingContent;
+import uk.gov.hmcts.cmc.claimstore.utils.DateUtils;
+import uk.gov.hmcts.cmc.claimstore.utils.Formatting;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertReport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertRequest;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.RequireSupport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.UnavailableDate;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.cmc.ccd.util.StreamUtil.asStream;
+
+@Component
+public class HearingContentProvider {
+
+    private static final String DISABLED_ACCESS = "Disabled Access";
+    private static final String YES = "YES";
+    private static final String NO = "NO";
+
+    private Function<ExpertReport, ExpertReportContent> mapExpertReport = report ->
+        ExpertReportContent.builder().expertName(report.getExpertName())
+            .expertReportDate(Formatting.formatDate(report.getExpertReportDate()))
+            .build();
+    private Function<UnavailableDate, String> mapToISOFullStyle = unavailableDate ->
+        Optional.ofNullable(unavailableDate)
+            .map(UnavailableDate::getUnavailableDate)
+            .map(DateUtils::toISOFullStyle).orElse("");
+
+    private BiConsumer<RequireSupport, HearingContent.HearingContentBuilder> mapSupportRequirement =
+        (support, builder) -> {
```

Comment:
I don't particularly like this "building by side effect" pattern, but I understand we have it everywhere so it's ok :D

Reply:
Lol. You are rt. There are places where they call a method to return consumer. :-). But I didnt like the existing approach, So I used the Biconsumer.

Reply:
This BiConsumer is not used as the assignment target for a lambda expression or method reference. It's not a valid use case for BiConsumer here.